src/workspacer.Bar/Widgets/TitleWidget.cs

```diff
@@ -1,51 +1,38 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace workspacer.Bar.Widgets
 {
     public class TitleWidget : BarWidgetBase
     {
-        public Color MonitorHasFocusColor { get; set; } = Color.Yellow;
+        #region Properties
+        public Color WindowHasFocusColor { get; set; } = Color.Yellow;
         public bool IsShortTitle { get; set; } = false;
+        public int? MaxTitleLength { get; set; } = null;
+        public bool ShowAllWindowTitles { get; set; } = false;
+        public string TitlePreamble { get; set; } = null;
+        public string TitlePostamble { get; set; } = null;
         public string NoWindowMessage { get; set; } = "No Windows";
-
-
-        public override IBarWidgetPart[] GetParts()
-        {
-            var window = GetWindow();
-            var isFocusedMonitor = Context.MonitorContainer.FocusedMonitor == Context.Monitor;
-            var multipleMonitors = Context.MonitorContainer.NumMonitors > 1;
-            var color = isFocusedMonitor && multipleMonitors ? MonitorHasFocusColor : null;
-
-            if (window != null)
-            {
-                if (!IsShortTitle)
-                {
-                    return Parts(Part(window.Title, color, fontname: FontName));
-                }
-                else
-                {
-                    var shortTitle = GetShortTitle(window.Title);
-                    return Parts(Part(shortTitle, color, fontname: FontName));
-                }
-            }
-            else
-            {
-                return Parts(Part(NoWindowMessage, color, fontname: FontName));
-            }
-        }
+        public Func<IWindow, Action> TitlePartClicked = ClickAction;
+        public Func<IWindow, object> OrderWindowsBy = (window) => { return $"{window.IsFocused}{window.Title}"; };
```

Comment:
I'm not sure what other WMs do, but to me, as this has some resemblance to `WorkspaceWidget`, I would (perhaps naively) expect the ordering of the window titles to reflect the current layout's stack and stay static.

```suggestion
        public Func<IWindow, object> OrderWindowsBy = (window) => 0;
```

Reply:
I agree that the default of unsorted is better and is the more default. Allowing it to be customized i forgot to set it to static.